### ISSUE TITLE
NULL options are removed with higher-level access

### ISSUE THREAD

**Issue by YasserHassan** [pyecharts/pyecharts#738]
The new `NULL` value implemented from #703 is not preserved when set to a chart option, then that chart is used by another entity like when added to a grid. Please check the example below.
```
from pyecharts.utils import NULL
from pyecharts import Line, Grid

line = Line("Line")
attr = ['A', 'B', 'C']
line.add('Line1', attr, [1, 2, 3])
print(line._option['series'][0]['symbol'])  # prints 'emptyCircle'

line._option['series'][0]['symbol'] = NULL
print('Line symbol value from _option:', line._option['series'][0]['symbol'])  # Passport text
print('Line symbol value from options:', line.options['series'][0]['symbol'])  # None

grid = Grid()
grid.add(line, grid_top="10%")

print('Line symbol value from grid _option:', grid._option['series'][0]['symbol'])  # None
print('Line symbol value from grid options:', grid.options['series'][0]['symbol'])  # KeyError: 'symbol'
```
The last line gives key error because the line's `options` first converted `NULL` to `None` by `remove_key_with_none_value()` then the new value was passed up to the grid which found `None` so it removed it when `remove_key_with_none_value()` was invoked again.

**Comment by chfw** [comment]
Thanks. I see the point. Let me think about the proper fix.

**Comment by chfw** [comment]
by the way, you can do `from pyecharts import NULL`

**Comment by YasserHassan** [comment]
@chfw yes I expected that but wanted to follow the test way below just in case. https://github.com/pyecharts/pyecharts/blob/3d94acfbcb86286bbfa6b3676ec2b06e53031129/test/test_utils.py#L8-L14

**Comment by chfw** [comment]
NOT_SET is more functional but the dev teams thought NULL for null is what it meant. And when we say a key is None in options is to say: let Echarts to choose its default value. Neither True or False is enough.
